internal/exec/exec_test.go

```diff
@@ -0,0 +1,64 @@
+package exec
+
+import (
+	"os"
+	"testing"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+func TestExecProcess(t *testing.T) {
+	var procThreadSize uintptr
+	if err := winapi.InitializeProcThreadAttributeList(nil, 1, 0, &procThreadSize); err != nil && err != windows.E_NOT_SUFFICIENT_BUFFER {
+		t.Fatalf("failed to call InitializeProcThreadAttributeList: %s", err)
+	}
+
+	procHeap, err := winapi.GetProcessHeap()
+	if err != nil {
+		t.Fatalf("GetProcessHeap failed: %s", err)
+	}
+
+	attrList, err := winapi.HeapAlloc(procHeap, 0, procThreadSize)
+	if err != nil {
+		t.Fatalf("HeapAlloc failed: %v\n", err)
+	}
+	defer winapi.HeapFree(procHeap, 0, attrList)
+
+	var procAttrList *winapi.PROC_THREAD_ATTRIBUTE_LIST
+	procAttrList = (*winapi.PROC_THREAD_ATTRIBUTE_LIST)(unsafe.Pointer(attrList))
+	if err = winapi.InitializeProcThreadAttributeList(procAttrList, 1, 0, &procThreadSize); err != nil {
+		t.Fatalf("InitializeProcThreadAttributeList - second call failed: %v\n", err)
+	}
+
+	// This is useless but it tests all the win32 calls working. A better use for this test would be creating
+	// this process to be a child of a different process other than the one launching it but we can't gurantee
+	// any other processes running/get their pids.
+	pHandle, err := windows.OpenProcess(winapi.PROCESS_ALL_ACCESS, false, uint32(os.Getpid()))
+	uintPtrHandle := uintptr(pHandle)
+	if err := winapi.UpdateProcThreadAttribute(
```

Comment:
It would be nice to have a utility function that consumers (such as this test) can use to take a `[]PROC_THREAD_ATTRIBUTE` and get a `PROC_THREAD_ATTRIBUTE_LIST`.

Reply:
Are you referring to the constants that are all prefixed with PROC_THREAD_ATTRIBUTE_* or the struct itself? If the former, I have this locally right now. If the latter, I don't know if passing in a slice of PROC_THREAD_ATTRIBUTE structs would be the best way to go about this. These aren't actually crafted by the caller nor are they accessed anywhere or in anyway by these calls.

Reply:
My feedback here is basically "constructing a `PROC_THREAD_ATTRIBUTE_LIST` requires a non-trivial amount of code, we should make this easier". I would encourage you to think about what such a helper function would look like, and see what we can come up with.

Reply:
All the user seems to care about is a set of (attribute, value, size) tuples, and that they get a `PROC_THREAD_ATTRIBUTE_LIST` as output.

Reply:
Agreed, for just one test it's ok but I have 3+ locally and the eyesore starts to show. I'll update this when I push and tell me what you think

Reply:
Tell me if the helper func is sufficient, it initializes the list for now but no updates. I was having trouble figuring out a nice way to setup the different UpdateProcThreadAttribute calls and how to do cleanup as the three scenarios we have all have differing cleanup so it was hard to make it generic in what we return for this based on what attributes were passed in.